source/agora/consensus/data/TransactionData.d

```diff
@@ -0,0 +1,297 @@
+        return buffer.idup;
+    }
+
+    /// Support for Vibe.d deserialization
+    public static TransactionData fromString (scope const(char)[] str) @safe
+    {
+        if (str.length % 2)
+            return TransactionData([], "The length of the hexadecimal string is not even");
+
+        if (str.length >= 2 && str[0] == '0' && (str[1] == 'x' || str[1] == 'X'))
+            str = str[2 .. $];
+
+        auto hexDecoder = (char c) @safe nothrow @nogc
+        {
+            switch (c) {
+                case '0': .. case '9':
+                    return cast(ubyte)(c - '0');
+                case 'a': .. case 'f':
+                    return cast(ubyte)(10 + c - 'a');
+                case 'A': .. case 'F':
+                    return cast(ubyte)(10 + c - 'A');
+                default:
+                    return 0xff;    //  mark error
+            }
+        };
+
+        ubyte [] data;
+        data.length = str.length / 2;
+
+        size_t idx;
+        foreach (chunk; str.byChar.map!(c => hexDecoder(c)).chunks(2))
+        {
+            if (chunk[0] == 0xff || chunk[1] == 0xff)
+                return TransactionData([], "The hexadecimal string contains an abnormal character");
+
+            data[idx++] = cast(ubyte)((chunk[0] << 4) + chunk[1]);
+        }
+
+        return TransactionData(data);
+    }
+
+    /***************************************************************************
+
+        Implements hashing support
+
+        Params:
+            dg = Hashing function accumulator
+
+    ***************************************************************************/
+
+    public void computeHash (scope HashDg dg) const nothrow @nogc @safe
+    {
+        hashPart(this.data, dg);
+    }
+
+    /***************************************************************************
+
+        TransactionData Serialization
+
+        Params:
+            dg = serialize function accumulator
+
+    ***************************************************************************/
+
+    public void serialize (scope SerializeDg dg) const @safe
+    {
+        serializePart(this.data, dg);
+    }
+
+    /***************************************************************************
+
+        Returns a new instance of type `TransactionData`
+
+        Params:
+            TransactionDataT = Qualified type of TransactionData to return
+            dg   = Delegate to read binary data
+            opts = Deserialization options (should be forwarded)
+
+        Returns:
+            A new instance of type `TransactionData`
+
+    ***************************************************************************/
+
+    public static TransactionDataT fromBinary (TransactionDataT = TransactionData) (
+        scope DeserializeDg dg, const ref DeserializerOptions opts) @safe
+    {
+        return TransactionData(deserializeFull!(ubyte[])(dg, opts));
+    }
+
+    /// Support for comparison
+    public int opCmp (ref const typeof(this) s) const nothrow @safe @nogc
```

Comment:
Why do we need comparison support?

Reply:
There is no particular reason, so I will remove it